pkg/trace/writer/payload.go

```diff
@@ -165,19 +179,24 @@ func (s *queuableSender) Run() {
 	for {
 		select {
 		case payload := <-s.in:
-			if stats, err := s.sendOrQueue(payload); err != nil {
-				log.Debugf("Error while sending or queueing payload. err=%v", err)
-				s.notifyError(payload, err, stats)
-			}
+			// enforce max inflight payloads with way to wait for them all to finish
+			s.inflightPayloads <- struct{}{}
```

Comment:
I'm not sure if this is the right approach. Let's discuss it a bit later when you're back.